benchmarks/314.hadoop/hadoop.sh

```diff
@@ -20,63 +17,53 @@ HIBENCH_BENCH_LIST=$HIBENCH_CONF/benchmarks.lst
 HIBENCH_LAN_API=$HIBENCH_CONF/languages.lst
 
 sudo apt-get install expect -y
+
 ##### set the ssh no-passwd login #####
 if [ ! -f ~/.ssh/*.pub ]; then
     ./generate_keys.sh
 fi
-public_key=$(cat ~/.ssh/*.pub)
-echo $public_key >> ~/.ssh/authorized_keys
+sKeyPub=$(cat ~/.ssh/id_*.pub)
+f0=~/.ssh/authorized_keys
+grep -q "${sKeyPub}" ${f0} 2>/dev/null
+if [ $? -ne 0 ]; then
+    cat ~/.ssh/*.pub >>${f0}
+fi
 
 ############## get JAVA_HOME which need to be used later ################
-java_version=`java -version 2>tmp_001.txt && awk '/java/{print $0}' tmp_001.txt`
-if [ "$java_version"x = ""x ]; then
-    sudo apt-get install openjdk-7-jdk -y
-    if [ $? -ne 0 ]; then
-        echo 'Install openjdk-7-jdk error'
-    fi
-    java_location=$(find /usr/lib -name 'java-*-openjdk*')
-    java_loc=$(echo $java_location | awk '{print $1}')
-    echo "java install location is $java_loc"
-
-    echo "export JAVA_HOME=$java_loc" >> /etc/profile
-    echo 'export JAVA_JRE=$JAVA_HOME/jre'  >> /etc/profile
-    echo 'CLASSPATH=.:$JAVA_HOME/lib:$JRE_HOME/lib:$CLASSPATH'  >> /etc/profile 
-    source /etc/profile
-elif [ "$(whereis java)"x != ""x ]; then 
-    java_location=$(find /usr/lib -name 'java-*-openjdk*')
-    java_loc=$(echo $java_location | awk '{print $1}')
-    echo "java install location is $java_loc"
-
-    echo "export JAVA_HOME=$java_loc" >> /etc/profile
-    echo 'export JAVA_JRE=$JAVA_HOME/jre'  >> /etc/profile
-    echo 'CLASSPATH=.:$JAVA_HOME/lib:$JRE_HOME/lib:$CLASSPATH'  >> /etc/profile 
-    source /etc/profile
-else
-    # java is installed 
-    flag=0
-    sys_path=(${PATH//:/ })
-    
-    for i in ${sys_path[@]}
-    do
-        tmp=$(echo $i | grep 'java' | grep 'bin' | grep -v 'jre')
-        if [ "$tmp"x != ""x ]; then
-            flag=1
-            java_loc=${i%bin}
+#[ "$(whereis java)"x != ""x ]
+if false; then
```

Comment:
if false; then -> This will evaluate to Boolean false, then the statements with in the if block always will be redundant. Any reason for doing it?